### PR TITLE
Better message for case incompatible package found

### PR DESCRIPTION
**Pull request by jgsogo**
Changelog: omit
Docs: https://github.com/conan-io/docs/pull/1910

After comment here https://github.com/conan-io/conan-center-index/issues/3379#issuecomment-720401803 , trying to make the message more explicit (not sure if we should add the recommendation: remove existing package)

### CODE REVIEW

File: conans/test/functional/case_sensitive_test.py [hunk 1]
```diff
@@ -1,24 +1,26 @@
...
-        self.output.info("Running source!")
-'''
+        def source(self):
+            self.output.info("Running source!")
+''')
 
 
 class CaseSensitiveTest(unittest.TestCase):
 
+    @unittest.skipIf(platform.system() == 'Linux', "Only for case sensitive OS")
```
Comment (memsharded) [inline conans/test/functional/case_sensitive_test.py]: Skip message is wrong, it is the opposite

File: conans/test/functional/command/remove_test.py [hunk 2]
```diff
@@ -14,12 +11,13 @@
...
 from conans.server.store.server_store import ServerStore
 from conans.test.assets.cpp_test_files import cpp_hello_conan_files
+from conans.test.utils.mocks import TestBufferConanOutput
 from conans.test.utils.test_files import temp_folder
 from conans.test.utils.tools import NO_SETTINGS_PACKAGE_ID, TestClient, \
     TestServer, GenConanfile
-from conans.test.utils.mocks import TestBufferConanOutput
 from conans.util.env_reader import get_env
 from conans.util.files import load
+from mock import Mock
```
Comment (memsharded) [inline conans/test/functional/command/remove_test.py]: Incorrect position for import

File: conans/test/functional/case_sensitive_test.py (hunk 1 above)
Comment (memsharded) [reply conans/test/functional/case_sensitive_test.py]: Maybe move the skipIf to class level?

**Review by danimtb — APPROVED** [review]

**Review by memsharded — APPROVED** [review]

**Comment by memsharded** [comment]
> (not sure if we should add the recommendation: remove existing package)

Probably yes, add this suggestion in the error message.

**Pull request merged**